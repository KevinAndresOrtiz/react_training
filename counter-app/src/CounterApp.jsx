import PropTypes from "prop-types";
import { useState } from "react";


export const CounterApp = ({ counter }) => {
    const [ counterValue, setCounterValue ] = useState(counter);
    const handleAdd = ()=> {
        setCounterValue(counterValue + 1);
    }

    const handleDecrease = () => {
        setCounterValue(counterValue - 1);
    }

    const handleReset = () => {
        setCounterValue(counter);
    }

    return(
        <>
        <h2>Counter App</h2>
        <p>{ counterValue }</p>
        <button onClick={ handleAdd }>
            +1
        </button>
        <button onClick={ handleDecrease }>
            -1
        </button>
        <button onClick={ handleReset }>
            Reset
        </button>
        </>
    )
};

CounterApp.propTypes = {
    counter: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    counter : 0
}