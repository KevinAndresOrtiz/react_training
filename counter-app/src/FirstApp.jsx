import PropTypes  from "prop-types";
import { Fragment } from "react";


export const App = ( { title, subtitle  } ) => {
    return (
        <Fragment>
            {title}
            <p>{ subtitle }</p>
        </Fragment>
    );
}

App.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
};
