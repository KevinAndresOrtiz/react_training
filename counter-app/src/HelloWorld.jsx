export const HelloWorld = () => {
    return (
        <h2>Hello World this is my first message</h2>
    );
};