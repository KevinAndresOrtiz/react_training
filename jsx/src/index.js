import React from "react";
import ReactDOM  from "react-dom/client";

const el = document.getElementById('root');

const root = ReactDOM.createRoot(el);

function App(){
    const inputType = 'number';
    const message = 'Enter age';
    const minValue = 5;
    return (
            <>
            <input type={inputType} min={minValue} max={100} />
            <input type={inputType} min={minValue} max={100} list={[1, 2, 3]} style={{color: 'red'}} alt={message} />
            </>
            );
}

root.render(<App />);