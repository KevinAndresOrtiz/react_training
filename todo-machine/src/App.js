import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <TodoItem />
      <TodoItem />
      <TodoItem />
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Aprendamos React
        </a>
      </header>
    </div>
  );
}

function TodoItem() {
  return(
    <li>
      <span>V</span>
      <p>Llorar con la llorona</p>
      <span>X</span>
    </li>
  );
}

export default App;
