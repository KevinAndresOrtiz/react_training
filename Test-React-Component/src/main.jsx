import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
// import App from './App.jsx'
// import './index.css'
// import { FirstApp } from './FirstApp.jsx'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp counter={100} />
  </React.StrictMode>,
)
