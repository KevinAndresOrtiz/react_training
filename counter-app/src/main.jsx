import React from 'react';
import ReactDOM  from 'react-dom/client';
import { App } from './FirstApp';
// import { HelloWorld } from './HelloWorld';
import './style.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <App title='Hola Soy Goku'/>
    </React.StrictMode>
);

