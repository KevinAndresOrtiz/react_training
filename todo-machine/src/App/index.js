import './App.css';
import { TodoCounter } from '../TodoCounter';
import { TodoSearch } from '../TodoSearch';
import { TodoList } from '../TodoList';
import { TodoItem } from '../TodoItem';
import { CreateTodoButton } from '../CreateTodoButton';
import React, { useState } from 'react';
import {useLocalStorage} from './useLocalStorage';

// const defaultTodos = [
//   { text: 'Cortar cebolla', completed: true },
//   { text: 'Tomar el curso de react.js', completed: false },
//   { text: 'Llorar con la llorona', completed: false },
//   { text: 'Curso 2 de react.js', completed: false },
//   { text: 'Tomar cursos de cloudguru', completed: false },
// ];


function App() {
  const [todos, saveTodos] = useLocalStorage('TODOS_V1', []);
  const [searchValue, setSearchValue] = useState('');
  const completedTodos = todos.filter((value) => value.completed === true).length;
  const totalTodos = todos.length;
  const searchedTodos = todos.filter((value) => value.text.toLowerCase().includes(searchValue.toLocaleLowerCase()));

  const completeTodos = (text) => {
    const newTodos = [...todos];
    const todoIndex = todos.findIndex((todo) => todo.text === text);
    newTodos[todoIndex].completed = true;
    saveTodos(newTodos)
  };

  const deleteTodos = (text) => {
    const todosDeleted = todos.filter((todo) => todo.text !== text);
    saveTodos(todosDeleted);
  };

  return (
    <React.Fragment>
      <TodoCounter completed={completedTodos} total={totalTodos} />
      <TodoSearch 
            searchValue={searchValue} 
            setSearchValue={setSearchValue}/>
      <TodoList >
        {searchedTodos.map(todo => (
          <TodoItem 
              key={todo.text} 
              text={todo.text} 
              completed={todo.completed} 
              onComplete={() => completeTodos(todo.text)}
              onDelete={() => deleteTodos(todo.text)}
          />)
          )
        }
      </TodoList>
      <CreateTodoButton />
    </React.Fragment>
  );
}


export default App;
