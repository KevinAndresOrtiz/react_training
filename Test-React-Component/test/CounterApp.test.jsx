import { render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en <CounterApp />', ()=> {
    test('Debe hacerse match con el snapshot', () => {
        const counterValue = 100;
        const { container } = render(<CounterApp counter={counterValue}/>);
        expect(container).toMatchSnapshot();

    });

    test('debe mostrarse el valor del counter en el componente', () => {
        const counterValue = 100;
        render(<CounterApp counter={counterValue} />);
        expect(screen.getByText(counterValue)).toBeTruthy();
        expect(screen.getByText(counterValue).innerHTML).toBe(counterValue.toString());
    });
});