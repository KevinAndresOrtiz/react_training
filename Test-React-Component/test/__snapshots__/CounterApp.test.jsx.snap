// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe hacerse match con el snapshot 1`] = `
<div>
  <h2>
    Counter App
  </h2>
  <p>
    100
  </p>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
